PROTOCOL.md

```diff
@@ -1343,6 +1344,16 @@ When enabled:
  - The value of `delta.generationExpression` SHOULD be parsed as a SQL expression.
  - Writers MUST enforce that any data writing to the table satisfy the condition `(<value> <=> <generation expression>) IS TRUE`. `<=>` is the NULL-safe equal operator which performs an equality comparison like the `=` operator but returns `TRUE` rather than NULL if both operands are `NULL`
 
+## Default Columns
+
+Delta supports defining default expressions for columns on Delta tables. Delta will generate default values for columns when users do not explicitly provide values for them when writing to such tables. To enable Default Columns:
```

Comment:
Is there a allowed expressions list or is it dependent on a particular query engine?

Reply:
Good question. Like "Generated Columns", the metadata value is stored as a string intended to represent an arbitrary SQL expression.

The intent is for whichever engine performing writes to the table, if the rows are lacking values for those columns, the engine is supposed to parse and analyze the string as a SQL expression and evaluate it, then use the result for that column. By the same token, if the engine specified the explicit `DEFAULT` SQL keyword for any column, the expression result should be substituted in the same way.

Any engine that assigns the default value can use its own SQL dialect of choice to represent the expression as a string, and use that same dialect to evaluate that expression later for future writes. If one engine writes the string metadata using its own SQL dialect and another engine then reads it back later when performing writes, the results are undefined. In practice, only Apache Spark can support reading and writing this metadata so far, so the question is moot, but in the future if another engine such as Presto added support for this feature, then it could write the metadata using its own dialect and then read it back again later.